File: genkey.cpp
#include <iostream>
#include <random>
#include <string>

using namespace std;

string genkey() {
    string lcnkey = "";
    for (int anu = 0; anu < 4; anu++) {
        string block = "";
        for (int j = 0; j < 4; j++) {
            string charSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
            char randomChar = charSet[rand() % charSet.length()];
            block += randomChar;
        }
        lcnkey += block + "-";
    }
    lcnkey = lcnkey.substr(0, lcnkey.length() - 1);
    return lcnkey;
}

int main() {
    cout << genkey() << endl;
    return 0;
}

File: genkey.go
package main

import (
	"fmt"
	"math/rand"
	"strings"
	"time"
)

func genkey() string {
	rand.Seed(time.Now().UnixNano())

	lcnkey := ""
	for anu := 0; anu < 4; anu++ {
		block := make([]byte, 4)
		for j := 0; j < 4; j++ {
			charSet := []byte(strings.ToUpper("ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"))
			block[j] = charSet[rand.Intn(len(charSet))]
		}
		lcnkey += string(block) + "-"
	}
	lcnkey = lcnkey[:len(lcnkey)-1]
	return lcnkey
}

func main() {
	fmt.Println(genkey())
}
